vespa-feed-client/src/test/java/ai/vespa/feed/client/examples/JsonStreamFeederExample.java

```diff
@@ -0,0 +1,146 @@
+                    continue;
+                log.info("Put document " + op.documentId);
+                CompletableFuture<Result> promise;
+                DocumentId docId = DocumentId.of(op.documentId);
+                OperationParameters params = OperationParameters.empty();
+                String json = op.documentFieldsJson;
+                switch (op.type) {
+                    case "put":
+                        promise = feedClient.put(docId, json, params);
+                        break;
+                    case "remove":
+                        promise = feedClient.remove(docId, params);
+                        break;
+                    case "update":
+                        promise = feedClient.update(docId, json, params);
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Invalid operation: " + op.type);
+                }
+                sentCounter.incrementAndGet();
+                promise.whenComplete((result, throwable) -> {
+                    if (resultCounter.getAndIncrement() % 10 == 0) {
+                        printProgress();
+                    }
+                    if (throwable != null) {
+                        failureCounter.incrementAndGet();
+                        System.err.println("Failure: " + throwable);
+                        throwable.printStackTrace();
+                    } else if (result.type() == Result.Type.failure) {
+                        failureCounter.incrementAndGet();
+                        System.err.println("Failure: " + result.resultMessage());
+                    }
+                });
+            } catch (InterruptedException e) {
+                log.log(Level.SEVERE, "Got interrupt exception.", e);
+                break;
+            }
+        }
+        log.info("Shutting down feeding thread");
+        this.feedClient.close();
+        finishedDraining.countDown();
+    }
+
+    void printProgress() {
+        synchronized (monitor) {
+            Instant now = Instant.now();
+            int resultCounter = this.resultCounter.get();
+            int failureCounter = this.failureCounter.get();
+            int sentCounter = this.sentCounter.get();
+            double docsDelta = resultCounter - failureCounter - startSampleResultCount;
+            Duration duration = Duration.between(startSampleInstant, now);
+            startSampleInstant = now;
+            this.startSampleResultCount = resultCounter - failureCounter;
+            long durationMilliSecs = duration.toMillis() + 1; // +1 to avoid division by zero
+            double rate = 1000. * docsDelta / durationMilliSecs;
+            System.err.println(new Date() + " Result received: " + resultCounter
+                    + " (" + failureCounter + " failed so far, " + sentCounter
+                    + " sent, success rate " + String.format(Locale.US, "%.2f docs/sec", rate) + ").");
+        }
+    }
+}
```

Comment:
Not sure I like this example very much. Could we rewrite it to just use the `FeedClient` directly? That one blocks when you attempt to send new operations and there's no free inflight slots, which avoids blowing up memory.

Reply:
Agreed - the example is now simplified to just print result of `FeedClient.stats()` instead of manual book-keeping.